internal/engine/worker/decentralized/contract/polymarket/worker.go

```diff
@@ -0,0 +1,221 @@
+package polymarket
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/rss3-network/node/config"
+	"github.com/rss3-network/node/internal/engine"
+	source "github.com/rss3-network/node/internal/engine/source/ethereum"
+	"github.com/rss3-network/node/provider/ethereum"
+	"github.com/rss3-network/node/provider/ethereum/contract"
+	"github.com/rss3-network/node/provider/ethereum/contract/polymarket"
+	"github.com/rss3-network/node/provider/ethereum/token"
+	"github.com/rss3-network/node/schema/worker/decentralized"
+	"github.com/rss3-network/protocol-go/schema"
+	activityx "github.com/rss3-network/protocol-go/schema/activity"
+	"github.com/rss3-network/protocol-go/schema/metadata"
+	"github.com/rss3-network/protocol-go/schema/network"
+	"github.com/rss3-network/protocol-go/schema/tag"
+	"github.com/rss3-network/protocol-go/schema/typex"
+	"github.com/samber/lo"
+	"github.com/shopspring/decimal"
+	"go.uber.org/zap"
+)
+
+var _ engine.Worker = (*worker)(nil)
+
+type worker struct {
+	ethereumClient ethereum.Client
+	tokenClient    token.Client
+	ctfExchange    *polymarket.CTFExchangeFilterer
+	// negRiskCTF     *polymarket.NegRiskCTFExchangeFilterer
+}
+
+func (w *worker) Name() string {
+	return decentralized.Polymarket.String()
+}
+
+func (w *worker) Platform() string {
+	return decentralized.PlatformPolymarket.String()
+}
+
+func (w *worker) Network() []network.Network {
+	return []network.Network{
+		network.Polygon,
+	}
+}
+
+func (w *worker) Tags() []tag.Tag {
+	return []tag.Tag{
+		tag.Collectible,
+	}
+}
+
+func (w *worker) Types() []schema.Type {
+	return []schema.Type{
+		typex.CollectibleTrade,
+	}
+}
+
+func (w *worker) Filter() engine.DataSourceFilter {
+	return &source.Filter{
+		LogAddresses: []common.Address{
+			polymarket.AddressPolyMarketCTFExchange,
+			polymarket.AddressPolyMarketNegRiskCTFExchange,
```

Comment:
i saw `negRiskCTF` is commented for now, do we still need filter its contract address?

Reply:
No, it's no longer needed. I removed`polymarket.AddressPolyMarketNegRiskCTFExchange` from the return value.

Reply:
Yes, I want to keep the address for matching. The reason why I commented out `negRiskCTF` is due to code redeclaration issues between the 2 contract generated files. But both 2 contracts share the same `ParseOrderFilled` function and its parameter the `ExchangeOrderFilled` struct. I decided to use the function of 'ctfExchange' only.

Reply:
ok, make sense